micrometer-jakarta/src/main/java/io/micrometer/jakarta/instrument/binder/http/ObservationAutoDiscoverable.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2023 VMware, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.jakarta.instrument.binder.http;
+
+import jakarta.ws.rs.ConstrainedTo;
+import jakarta.ws.rs.RuntimeType;
+import jakarta.ws.rs.core.FeatureContext;
+import org.glassfish.jersey.internal.spi.AutoDiscoverable;
+
+/**
+ * Binds observation into Jersey.
+ */
+@ConstrainedTo(RuntimeType.CLIENT)
+public class ObservationAutoDiscoverable implements AutoDiscoverable {
```

Comment:
I'm not familiar with how libraries provide these `AutoDiscoverable` implementations, but the JavaDoc mentions declaring the implementation in a file in `META-INF/services` named `org.glassfish.jersey.internal.spi.AutoDiscoverable`. Otherwise I guess this won't be auto-discovered, but as I said, I'm not familiar with this.

Looking at [the documentation](https://eclipse-ee4j.github.io/jersey.github.io/documentation/latest/deployment.html#deployment.autodiscoverable), it says this is internal, so I wonder if we should be providing this implementation at all.

> Auto discovery functionality is in Jersey supported by implementing an internal `AutoDiscoverable` Jersey SPI. This interface is not public at the moment, and is subject to change in the future, so be careful when trying to use it.